File: app/src/main/java/com/john/kot/mvvm/viewmodel/UserModel.java
package com.john.kot.mvvm.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.LinkedList;

public class UserModel extends ViewModel {

    private final MutableLiveData userLiveData = new MutableLiveData();
    private final LinkedList<User> userList = new LinkedList<>();

    public LiveData getUser() {
        return userLiveData;
    }

    public UserModel() {
    }

    void doAction() {
        userList.add(new User("lili"));
        userLiveData.postValue(userList);
    }
}

File: app/src/main/java/com/john/kot/mvvm/viewmodel/UserModel.kt
package com.john.kot.mvvm.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.util.*

class UserModel : ViewModel() {
    private val userLiveData = MutableLiveData<List<UserObject>>()
    private val userList = LinkedList<UserObject>()
    val user: LiveData<List<UserObject>>
        get() = userLiveData

    fun doAction() {
        userList.add(UserObject("lili"))
        userLiveData.postValue(userList)
    }
}